src/Microsoft.TemplateEngine.Abstractions/ICacheTag.cs

```diff
@@ -1,13 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
 using System.Collections.Generic;
 
 namespace Microsoft.TemplateEngine.Abstractions
 {
+    /// <summary>
+    /// Represents a parameter symbol with choices in template cache file.
+    /// </summary>
     public interface ICacheTag
     {
-        string Description { get; }
+        /// <summary>
+        /// Gets the friendly name of the tag to be displayed to the user.
+        /// </summary>
+        string? DisplayName { get; }
+
+        /// <summary>
+        /// Gets the description of the tag to be displayed to the user.
+        /// </summary>
+        string? Description { get; }
 
-        IReadOnlyDictionary<string, string> ChoicesAndDescriptions { get; }
+        IReadOnlyDictionary<string, ParameterChoice> Choices { get; }
```

Comment:
Visual Studio and  Mono uses this interface, I'm not sure about changing it

http://index/?leftProject=Microsoft.TemplateEngine.Abstractions&leftSymbol=kzkyqi5zfqek

Reply:
Actually just realized further impact - `TemplateInfo` is also used in Template Discovery tool. If we rename it now, the file generated by the tool won't be compatible with previous versions.

Reply:
Since we moved this work to .NET 6,  can we keep this change?